File: Dataset/Leetcode/train/70/126.java
class Solution {
    public int XXX(int n) {
        if(n<3){
            return n;
        }
        int[] array = new int[n+1];
        array[0] = 0;
        array[1] = 1;
        array[2] = 2;
        for(int i = 3;i<n+1;i++){
            array[i] = array[i-1]+array[i-2];
        }
        return array[n];
    }
}

File: Dataset/Leetcode/train/78/126.py
class Solution(object):
    def XXX(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        #子集个数为2**n, n 为nums里面的元素个数
        ret = []
        
        for item in nums:
            ret.append([item])
            for i in range(len(ret) - 1):
                # temp = ret[i] 浅拷贝
                temp = ret[i][:] #深拷贝
                temp.append(item)
                ret.append(temp)
        ret.append([])
        return ret

